### PR TITLE
Ring bell when is hit by a projectile

### PR DESCRIPTION
**Pull request by IvanCraft623**
## Introduction

As title, I would like to target this to next-minor, but because `Block::onProjectileHit()` only exists in next-major it is not possible.

### CODE REVIEW

File: src/block/Bell.php [hunk 1]
```diff
@@ -152,6 +154,22 @@ public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player
 		return true;
 	}
 
+	public function onProjectileHit(Projectile $projectile, RayTraceResult $hitResult) : void{
+		$faceHit = $hitResult->getHitFace();
+		if($this->attachmentType->equals(BellAttachmentType::CEILING())){
+			$this->ring($faceHit);
+		}
+		if($this->attachmentType->equals(BellAttachmentType::FLOOR()) && Facing::axis($faceHit) === Facing::axis($this->facing)){
```
Comment (dktapps) [inline src/block/Bell.php]: This should be using `elseif`

File: src/block/Bell.php [hunk 2]
```diff
@@ -152,6 +154,22 @@ public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player
...
+			$this->ring($faceHit);
+		}
+		if(
```
Comment (dktapps) [inline src/block/Bell.php]: same again here

Comment (dktapps) [reply src/block/Bell.php]: In fact all of these conditions can be merged, since the bodies of all of them are identical.

**Review by dktapps — CHANGES_REQUESTED** [review]

**Comment by dktapps** [comment]
The base code was changed, you'll need to merge changes.

**Comment by IvanCraft623** [comment]
> The base code was changed, you'll need to merge changes.

done

File: src/block/Bell.php [hunk 3]
```diff
@@ -158,4 +160,16 @@ public function ring(int $faceHit) : void{
...
+
+	private function isValidFaceToRing(int $faceHit) : bool{
+		return (
+			$this->attachmentType->equals(BellAttachmentType::CEILING()) ||
+			($this->attachmentType->equals(BellAttachmentType::FLOOR()) && Facing::axis($faceHit) === Facing::axis($this->facing)) ||
+			(
+				$this->attachmentType->equals(BellAttachmentType::ONE_WALL()) ||
+				$this->attachmentType->equals(BellAttachmentType::TWO_WALLS())) &&
+				($faceHit === Facing::rotateY($this->facing, false) || $faceHit === Facing::rotateY($this->facing, true)
+			)
```
Comment (dktapps) [inline src/block/Bell.php:171]: The braces in this block are a mess. I can't make head or tail of what's going on.

Comment (IvanCraft623) [reply src/block/Bell.php:171]: It's exactly the code you wrote

Comment (dktapps) [reply src/block/Bell.php:171]: My code:
image
Your code:
image

I don't know what happened to the formatting here, but it's a confusing mess.

Comment (IvanCraft623) [reply src/block/Bell.php:171]: Your code has been successfully copied and pasted

**Pull request merged**